Tests/Common/Scheduling/DateRulesTests.cs

```diff
@@ -313,6 +313,239 @@ public void EndOfMonthWithSymbolWithOffset(Symbols.SymbolsKey symbolKey, int[] e
             }
         }
 
+        [Test]
+        public void StartOfYearNoSymbol()
+        {
+            var rules = GetDateRules();
+            var rule = rules.YearStart();
+            var dates = rule.GetDates(new DateTime(2000, 01, 01), new DateTime(2010, 12, 31));
+
+            int count = 0;
+            foreach (var date in dates)
+            {
+                count++;
+                Assert.IsTrue(date.Day <= 4);
+            }
+
+            Assert.AreEqual(11, count);
+        }
+
+        [Test]
+        public void StartOfYearNoSymbolMidYearStart()
+        {
+            var rules = GetDateRules();
+            var rule = rules.YearStart();
+            var dates = rule.GetDates(new DateTime(2000, 06, 01), new DateTime(2010, 12, 31));
+
+            int count = 0;
+            foreach (var date in dates)
+            {
+                count++;
+                Assert.AreEqual(1, date.Day);
+            }
+
+            Assert.AreEqual(10, count);
+        }
+
+        [Test]
+        public void StartOfYearNoSymbolWithOffset()
+        {
+            var rules = GetDateRules();
+            var rule = rules.YearStart(5);
+            var dates = rule.GetDates(new DateTime(2000, 01, 01), new DateTime(2010, 12, 31));
+
+            int count = 0;
+            foreach (var date in dates)
+            {
+                count++;
+                Assert.AreEqual(6, date.Day);
+            }
+            Assert.AreEqual(11, count);
+        }
+
+        [TestCase(2, false)]       // Before 11th
+        [TestCase(4, false)]
+        [TestCase(8, false)]
+        [TestCase(12, true)]      // After 11th
+        [TestCase(16, true)]
+        [TestCase(20, true)]
+        public void StartOfYearSameYearSchedule(int startingDateDay, bool expectNone)
+        {
+            var startingDate = new DateTime(2000, 1, startingDateDay);
+            var endingDate = new DateTime(2000, 12, 31);
+
+            var rules = GetDateRules();
+            var rule = rules.YearStart(10); // 11/1/2000
+            var dates = rule.GetDates(startingDate, endingDate);
+
+            Assert.AreEqual(expectNone, dates.IsNullOrEmpty());
+
+            if (!expectNone)
+            {
+                Assert.AreEqual(new DateTime(2000, 1, 11), dates.First());
```

Comment:
Could be `dates.Single()` just to make sure.
And I guess you could ditch the `expectNone` argument and just check `startingDateDay > 11`.